### PR TITLE
2D slicing in MovieJSONGenerator when single movie is cached

### PR DESCRIPTION
**Pull request by mikejhuang**

### CODE REVIEW

**Review by aamster — CHANGES_REQUESTED** [review]
Could you please add a test showing that the for loop and the vectorized implementation are equivalent?

**Comment by mikejhuang** [comment]
> Could you please add a test showing that the for loop and the vectorized implementation are equivalent?

The test you added that tests if the cached movie and the non-cached movie outputs are equivalent is also used to test the 2D slicing since the 2D slicing logic is only applied when the movie is cached. 

Granted, there are ways the outputs aren't equivalent that the assert checks wouldn't catch.

**Comment by aamster** [comment]
I didn't see such a test. I'm looking for a test such as 
```
gen1 = MovieJSONGenerator(..., movs=...)
gen2 = MovieJSONGenerator(..., movs=None)
assert gen1[0] == gen2[0]
```

File: tests/test_generator_collection.py [hunk 1]
```diff
@@ -106,8 +106,15 @@ def test__getitem__(
         )
         batch = gen[0]
         assert len(batch) == 2 and \
-               batch[0].shape[0] > 0 and \
-               batch[1] .shape[0] > 0
+            batch[0].shape[0] > 0 and \
+            batch[1] .shape[0] > 0
+        if movs:
```
Comment (aamster) [inline tests/test_generator_collection.py]: Can you please make this a separate test?

**Review by aamster — APPROVED** [review]

**Pull request merged**